tests/functional/api/annotations_test.py

```diff
@@ -168,6 +170,116 @@ def test_it_updates_annotation_if_authorized(
         assert res.json["text"] == "This is an updated annotation"
         assert res.status_code == 200
 
+    # TODO Private annotations don't always have the NULL state: if a shared annotation is edited and made private it retains its previous state from when it was shared (Pending, Approved, Denied, or Spam).
+    @pytest.mark.parametrize(
+        "pre_moderation_enabled,moderation_status,expected_moderation_status",
+        [
+            # Editing a private annotation and making it shared might change its annotation state, depending on the current state:
+            # If the group has pre-moderation disabled the annotation's moderation state becomes Approved.
+            # If the group has pre-moderation enabled the annotation's moderation state becomes Pending.
+            # (So it's the same as what happens when creating a new shared annotation, but the state change only happens when the private annotation is made shared. This is important because the state change depends on the group's pre-moderation setting at the moment when the annotation becomes shared.)
+            (False, None, ModerationStatus.APPROVED),
+            (True, None, ModerationStatus.PENDING),
+            # If a private annotation whose state is Denied is edited and made shared the state becomes Pending.
+            # (Same as what happens when editing a Denied, shared annotation, but the state change only happens when the private annotation is made shared.)
+            (True, ModerationStatus.DENIED, ModerationStatus.PENDING),
+            # TODO WHAT ABOUT MODEREATED DISABLED?
+            # If a private annotation whose state is Pending or Spam is edited and made shared the state doesn't change.
+            # (Same as when editing a Pending or Spam, shared annotation.)
+            (True, ModerationStatus.PENDING, ModerationStatus.PENDING),
+            (True, ModerationStatus.SPAM, ModerationStatus.SPAM),
+        ],
+    )
+    def test_sharing_a_private_annotation(
```

Comment:
I'm using func test for this cases that require a visibility change.

This would be the edit view, calling the anno write service and that calling the moderation service.

These same cases need to be unit tested but I wanted to see if everything was wired correctly.